pkg/agent/client.go

```diff
@@ -248,6 +258,14 @@ func (a *AgentClient) initializeAuthContext(ctx context.Context) (context.Contex
 // The requests include things like opening a connection to a server,
 // streaming data and close the connection.
 func (a *AgentClient) Serve() {
+	defer func() {
+		// close all of conns with remote when AgentClient closed
+		for _, connCtx := range a.connManager.List() {
+			connCtx.cleanup()
```

Comment:
Can we delete the connCtx from the connManager. Otherwise we end up with a broken connection dangling in the connection list. While we are on the way out, it still seems prone to race conditions.

Reply:
The cleanup function automatically deletes the connection from the list. (https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/ff2a951482c1cdc615e00107fac9a42bf279f042/pkg/agent/client.go#L341)

Reply:
Good catch. I withdraw the need for the delete. I think the RLock still applies though.